src/bindings/python/flux/accounting/accounting_cli_functions.py

```diff
@@ -66,15 +66,56 @@ def view_bank(conn, bank):
 
 
 def delete_bank(conn, bank):
-    last_parent_bank_seen = bank
-
-    # delete bank from bank_table
-    delete_stmt = "DELETE FROM bank_table WHERE bank=?"
-    cursor = conn.cursor()
-    cursor.execute(delete_stmt, (bank,))
+    try:
+        # delete the first bank passed in
+        delete_stmt = """
+            DELETE FROM bank_table
+            WHERE bank=?
+            """
+        cursor = conn.cursor()
+        cursor.execute(delete_stmt, (bank,))
+
+        # construct a DataFrame object out of the
+        # bank passed in
+        df = pd.DataFrame([bank], columns=["bank"])
+        bank = df.iloc[0]
+
+        # helper function to traverse the bank table
+        # and delete all of its sub banks
+        def get_sub_banks(row):
+            select_stmt = """
+                SELECT bank
+                FROM bank_table
+                WHERE parent_bank=?
+                """
+            cursor = conn.cursor()
+            dataframe = pd.read_sql_query(select_stmt, conn, params=(row["bank"],))
+
+            # we've reached a bank with no sub banks
+            if len(dataframe) == 0:
+                select_associations_stmt = """
+                    SELECT user_name, bank
+                    FROM association_table
+                    WHERE bank=?
+                    """
+                for row in cursor.execute(select_associations_stmt, (row["bank"],)):
+                    delete_user(conn, user=row[0], bank=row[1])
```

Comment:
`delete_user` calls `conn.commit`, which will prevent an effective rollback in the case of an exception.  If `delete_user` is only called by this function, then I'd remove the `conn.commit()` call.  If it is called by other functions, maybe add an optional argument `commit` to allow committing to be turned off for this particular call?

Reply:
Ah, good point. I don't believe `delete_user` is called by anything else, so I'll go ahead and remove that `commit()` call. Thanks for pointing that out.

Reply:
Hmmm. I'm still seeing `conn.commit` in `delete_user` in the diff.  Maybe just a new push is needed?

Reply:
That's a fail on my part. Made the change on my machine but didn't push it. Sorry about that!